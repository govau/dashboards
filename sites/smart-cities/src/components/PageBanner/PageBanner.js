import React from 'react';
import PropTypes from 'prop-types';
import IndicatorCard from '../IndicatorCard/IndicatorCard';
import {
  INDICATORS,
  ELEMENT_IDS,
} from '../../constants';
import getColorVariant from '../../helpers/getColorVariant';
import style from './PageBanner.scss';

const PageBanner = props => (
  <div style={{ backgroundColor: getColorVariant(props.colorName, '020') }}>
    <div className={style.container}>
      <div className={style.textAndIndicator}>
        <div className={style.titleText}>
          <h1
            id={ELEMENT_IDS.INITIAL_FOCUS_POINT_ON_PAGE}
            className={style.title}
          >
            {props.title}

            {props.isAllCitiesCategory && (
              <span className={style.subTitle}> All cities</span>
            )}
          </h1>

          <p className={style.description}>{props.description}</p>

          {props.isAllCitiesCategory && (
            <div className={style.featuredTextWrapper}>
              <span className={style.featuredText}>Featured indicators</span>

              <span> Showing range</span>
            </div>
          )}
        </div>

        <div>
          {props.indicatorSubCategory && (
            <div className={style.heroSubCategoryWrapper}>
              <span className={style.heroSubCategoryText}>
                Key indicator
              </span>

              <span> {props.indicatorSubCategory}</span>
            </div>
          )}

          <IndicatorCard
            size="large"
            className={style.card}
            color={getColorVariant(props.colorName, '500')}
            colorName={props.colorName}
            indicatorId={props.indicatorId}
            cities={props.cities}
            city={props.city}
          />
        </div>
      </div>
    </div>
  </div>
);

const cityPropShape = {
  id: PropTypes.string.isRequired,
  indicators: PropTypes.object.isRequired,
  name: PropTypes.string.isRequired,
};

PageBanner.propTypes = {
  cities: PropTypes.arrayOf(PropTypes.shape(cityPropShape)),
  city: PropTypes.shape(cityPropShape),
  colorName: PropTypes.string.isRequired,
  description: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object, // city descriptions can be JSX
  ]).isRequired,
  indicatorSubCategory: PropTypes.string,
  indicatorId: PropTypes.oneOf(Object.keys(INDICATORS)).isRequired,
  title: PropTypes.string.isRequired,
  isAllCitiesCategory: PropTypes.bool,
};

export default PageBanner;
