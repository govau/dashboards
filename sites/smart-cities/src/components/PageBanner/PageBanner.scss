@import '~style/main';

.container {
  @include container;

  padding-top: 16px;

  @include for-tablet-portrait-up {
    padding-top: 32px;
  }
}

.textAndIndicator {
  display: flex;
  flex-direction: column;
  margin-bottom: 16px;

  @include for-tablet-portrait-up {
    flex-direction: row;
    margin-bottom: 32px;
  }
}

.titleText {
  margin-bottom: 32px;

  @include for-tablet-portrait-up {
    margin-bottom: 0;
    margin-right: 32px;
  }
}

.title {
  margin-bottom: 16px;
  font-size: 30px;
  font-weight: 500;
  line-height: 1.2;

  @include for-tablet-portrait-up {
    font-size: 36px;
  }

  @include for-tablet-landscape-up {
    font-size: 44px;
  }
}

.subTitle {
  font-weight: 300;
}

.description {
  font-size: 20px;
  font-weight: 500;
  line-height: 1.4;

  @include for-tablet-landscape-up {
    font-size: 22px;
    font-weight: 300;
  }
}

.featuredTextWrapper {
  padding-top: 32px;
  font-size: 20px;
  font-weight: 300;
  letter-spacing: 1.1px;
}

.featuredText {
  margin-right: 6px;
  font-weight: 500;
}

.heroSubCategoryWrapper {
  padding-top: 28px;
  margin-bottom: 16px;
  font-size: 14px;
}

.heroSubCategoryText {
  font-weight: 500;
}

.card {
  width: 100%;

  @include for-tablet-portrait-up {
    flex: none;
    width: 256px;
    margin-left: auto; // push to the right
  }
}

.indicatorTypeMarkWrapper {
  display: flex;
  align-items: center;
}

.indicatorTypeMark {
  margin-right: 8px;
}
