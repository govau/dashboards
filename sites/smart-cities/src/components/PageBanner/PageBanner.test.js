import React from 'react';
import { shallow } from 'enzyme';
import PageBanner from './PageBanner';

jest.mock('../IndicatorCard/IndicatorCard', () => 'IndicatorCard');

const defaultProps = {
  colorName: 'jobs',
  description: 'This is the mock category',
  indicatorId: 'population',
  title: 'Banner title',
  cities: [
    {
      id: 'perth',
      name: 'Perth',
      indicators: {
        population: 77,
      }
    },
    {
      id: 'sydney',
      name: 'Sydney',
      indicators: {
        population: 23,
      }
    },
  ],
};

it('should match Snapshot', () => {
  const component = shallow(
    <PageBanner {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should set the background color', () => {
  const component = shallow(
    <PageBanner {...defaultProps} />
  );

  const componentStyle = component.find('div').at(0).prop('style');
  // note, here the color is the string 'JOBS_020' because of the way
  // styles are imported during tests, but in reality
  // this will be the hex color
  expect(componentStyle.backgroundColor).toBe('JOBS_020');
});
