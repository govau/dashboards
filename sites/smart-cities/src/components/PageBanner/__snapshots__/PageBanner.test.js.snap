// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div style={{...}}>
  <div className=\\"container\\">
    <div className=\\"textAndIndicator\\">
      <div className=\\"titleText\\">
        <h1 id=\\"focus-point\\" className=\\"title\\">
          Banner title
        </h1>
        <p className=\\"description\\">
          This is the mock category
        </p>
      </div>
      <div>
        <IndicatorCard size=\\"large\\" className=\\"card\\" color=\\"JOBS_500\\" colorName=\\"jobs\\" indicatorId=\\"population\\" cities={{...}} city={[undefined]} />
      </div>
    </div>
  </div>
</div>"
`;
